index.bs

```diff
@@ -358,6 +360,21 @@ It is defined by running the following steps.
 </dd>
 </dl>
 
+### Methods ### {#generator-methods}
+<dl>
+<dt><dfn method for=MediaStreamTrackGenerator>setMuted()</dfn></dt>
+<dd>Allows muting or unmuting all tracks connected to this
+{{MediaStreamTrackGenerator}}, possibly including the {{MediaStreamTrackGenerator}}
+itself.
+
+When this method is called, the UA MUST execute the following steps, taking
+a boolean |newState| as input:
+1. For each track `t` connected to |generator|, {{SetMuted|set the muted state}} of `t` to |newState|.
+
+The UA MAY at any time automatically {{SetMuted|set the muted state}} of any
+track connected to a {{MediaStreamTrackGenerator}} if it detects absence or
+presence of media data flowing through the track.
```

Comment:
I think it would be better if we only add this statement. If the UA already can change the muted state, what does setMuted actually do? Just provide a hint to the UA?

Reply:
The UA can always change the muted state. That's how muted is originally defined in mediacapture.
The idea here is to add a programatic way to do it to let the application implementing the source decide when to do it since this is a programatically defined source. This would be the equivalent of a hardware mute button on a microphone, but since this source is JS-defined, the mute button is also JS-defined.

The method in principle is an order to the UA. But it can also be interpreted as a hint, since the UA can arbitrarily change the state back right after the order.  The UA shouldn't do that without good cause, but it's theoretically possible.